Day 2/DayTwoPartOne.py

```diff
@@ -0,0 +1,43 @@
+import sys
+sys.path.insert(0, "C:/Users/Callipolis/Desktop/Programming/AoC2/Advent-of-Code-2015")
+from InputReader import InputReader
+
+input_reader = InputReader("input2.txt")
+raw_input = input_reader.get_list_with_lines()
+
+class WrapperOrder:
+    
+    paper_counter = 0
+    puzzle_input = None
+    
+    def __init__(self, puzzle_input):
+        self.puzzle_input = puzzle_input
+        self.format_lines()
+
+    def line_to_iterables(self, line):
+        nums = line.split("x")
+        return nums
+    
+    def get_material_per_box(self, line):
+        length = int(line[0])
+        width = int(line[1])
+        height = int(line[2])
+        side1 = length*width
+        side2 = width*height
+        side3 = height*length
+        line_total = 2*side1 + 2*side2 + 2*side3 + min(side1, side2, side3)
+        return line_total
+    
+    def format_lines(self):
+        for line in self.puzzle_input:
+            line = self.line_to_iterables(line)
```

Comment:
Nem talalok lassan helyet ahova odafoshatok :D 

```
for line in self.puzzle_input:
            line = self.line_to_iterables(line)
```

Generally nem jo otlet local valtozokat felulirni. Modositani fine, (mint a line_total += ...), de egy teljesen uj erteket adni neki confusing tud lenni. Itt is ugye az a baj, hogy a line nev fasza az elso sorban mert sorokon iteralsz, tokeletes. De aztan felul irod egy masik fogalommal, es utana mar mas nev kene neki.

Azert is szivas sokszor valtozot "ujrairni"  mert ha itt hozzaadsz egy uj sort ami az eredeti sor erteket szeretne a line-nak, akkor az az ertek elveszett mar mert felulirtad, nem fogod tudni a sort visszakapni mar. Mivel a neve misleading, lehet megorulsz hogy ja itt van, hasznalod, de az erteke mar nem ugyanaz mint ami a ciklus valtozoe lenne es kesz a bug.

Kulon evil amikor valtoztatod a tipusat. A python ezzel megengedo, Java/C# ezt nem is engedne. Egyik sorban meg string, masik sorban mar array. Nagyon bele lehet ebbe kavarodni, mert attol fugg hogy mi a tipusa hogy epp melyik sort nezed.

Furthermore, ciklus valtozokat never-ever valtoztass. Nagyon confusing bug-okhoz tudsz jutni vele. Egy foreach-nel meg fine, de egy `for(int i = 0; i < 10; i++)` eseten ha az i-t valtoztatgatod, akkor nem tudod ugye hogy hanyszor fog lefutni a ciklus. Akar vegtelen ciklusba is futhatsz, aztan orakba tellik kitalalni hogy mi van. Altalaban veletlenul szoktam ebbe belefutni, de akkor szidom magam rendesen.

Ebben a ket sorban megusztad, mert egy nagyon egyszeru kod bazissal dolgozol meg egyszeru problema, de nasty dolgok johetnek ki ebbol, szoval ne szokj hozza

Fix: hozz letre egy uj valtozot, dimensions = self.line_to_iterables(line)

Reply:
"Nem talalok lassan helyet ahova odafoshatok".
Proceeds to write 6 paragraphs. :D

Köszi, ez így érthető. Jelenleg:

`        for box in self.box_dimensions:
            dimensions = self.parse_dimensions(box)`
            
Aztán most szétbontom a Calculatort és csinálom a Factory-t, szóval ki tudja hogyan fog kinézni. :D